### PR TITLE
Update ccd data

### PR DESCRIPTION
**Pull request by qzhou-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

- Extract case types from DB
- Added fields requested by A&P team
- Added initialise flag to avoid load not initialised  caseTypes
- Update integration test to be build isolated
- Added Clock and set  default timezone

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/resources/application-test.yaml [hunk 1]
```diff
@@ -7,15 +7,24 @@ etl:
   connection-string: "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:52189/devstoreaccount1;"
 
 extraction:
+  max:
+    batch:
+      row: 1
+  filter:
+    jurisdiction:
+      in:
+      out:
```
Comment (qzhou-hmcts) [inline src/functionalTest/resources/application-test.yaml:16]: Filter to not extract these jurisdictions

File: src/functionalTest/resources/application-test.yaml [hunk 2]
```diff
@@ -7,15 +7,24 @@ etl:
   connection-string: "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:52189/devstoreaccount1;"
 
 extraction:
+  max:
+    batch:
+      row: 1
+  filter:
+    jurisdiction:
+      in:
```
Comment (qzhou-hmcts) [inline src/functionalTest/resources/application-test.yaml:15]: Jurisdictions to extract filter, this is to maintain backward compatibility  and avoid not initialised caseTypes

File: src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java [hunk 3]
```diff
@@ -40,81 +50,135 @@
     @Autowired
     private CaseDataService caseDataService;
 
-    @SuppressWarnings("PMD.CloseResource")
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java:43]: Code refactored to break in smaller pieces of code

File: src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java [hunk 4]
```diff
@@ -40,81 +50,135 @@
...
-                    Extractor extractor = extractorFactory.provide(extractionData.getType());
-                    ResultSet resultSet = executor.execute();
-                    String fileName = BlobFileUtils.getFileName(extractionData, toDate);
+    @Autowired
+    private Clock clock;
+
+    public void execute(boolean initialLoad) {
+
+        LocalDate now = LocalDate.now(clock);
+        List<CaseDefinition> caseDefinitions = caseDataService.getCaseDefinitions();
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java:59]: Loading case definitions present in DB

File: src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java [hunk 5]
```diff
@@ -40,81 +50,135 @@
...
+        log.info("Total case definitions loaded {}", caseDefinitions.size());
+        for (CaseDefinition caseDefinition : caseDefinitions) {
+            ExtractionData extractionData = getExtractionData(caseDefinition);
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java:62]: Override existing case type configuration to keep backward compatibility

File: src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java [hunk 6]
```diff
@@ -40,81 +50,135 @@
...
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private void processData(ExtractionData extractionData, LocalDate executionTime, boolean initialLoad) {
+
+        LocalDate lastUpdated = null;
+        try {
+            lastUpdated = getLastUpdateFromContainer(extractionData, initialLoad);
+        } catch (Exception e) {
+            log.error("Case type not initialised {}", extractionData.getCaseType());
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java:92]: Skipping not initialised case types

File: src/main/java/uk/gov/hmcts/reform/dataextractor/service/impl/ExtractorCsv.java [hunk 7]
```diff
@@ -9,19 +9,35 @@
...
-            printer.printRecords(resultSet);
+            count = printRecordsWithCounter(printer, resultSet);
             printer.flush();
         } catch (IOException | SQLException e) {
             throw new ExtractorException(e);
         }
+        return count;
     }
 
+    private int printRecordsWithCounter(final CSVPrinter printer, final ResultSet resultSet) throws SQLException, IOException {
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/service/impl/ExtractorCsv.java:30]: This is the same code as Csv library , I'm  just returning the lines  count

File: src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java [hunk 8]
```diff
@@ -40,81 +50,135 @@
...
+            return;
         }
 
+        LocalDate toDate = null;
+        QueryExecutor executor = null;
+        Extractor extractor = extractorFactory.provide(extractionData.getType());
+        final int extractionWindow = caseDataService.calculateExtractionWindow(extractionData.getCaseType(),
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java:99]: This windows is to try to spread evenly the initial load

File: src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java [hunk 9]
```diff
@@ -6,22 +6,32 @@
...
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 @Component
 @Slf4j
 public class ExtractionComponent {
 
+    private static final Output DEFAULT_OUTPUT = Output.JSON_LINES;
+    private static final Locale GB_LOCALE = Locale.ENGLISH;
+    //    private static final int DEFAULT_EXTRACTION_WINDOW = 30;
```
Comment (tchow8) [inline src/main/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponent.java:33]: Remove if not used?

File: src/test/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponentTest.java [hunk 10]
```diff
@@ -68,33 +76,47 @@
     @Mock
     private CaseDataServiceImpl caseDataService;
 
+    private final Clock clock = Clock.systemDefaultZone();
+
+    @BeforeEach
+    public void setUp() {
+        ReflectionTestUtils.setField(classToTest, "clock", clock);
```
Comment (tchow8) [inline src/test/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponentTest.java:83]: Would it work to spy on clock instead as its Autowired?

File: src/test/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponentTest.java [hunk 11]
```diff
@@ -68,35 +78,54 @@
     @Mock
     private CaseDataServiceImpl caseDataService;
 
+    @Mock
+    private Clock clock;
+
+    @BeforeEach
+    public void setUp() {
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
```
Comment (tchow8) [inline src/test/java/uk/gov/hmcts/reform/dataextractor/ExtractionComponentTest.java:86]: Need this if the clock getZone is mocked?

**Pull request merged**